### DISCUSSION TITLE
Change gathering to new system

### DISCUSSION THREAD

**Discussion by thebonbon** [EveronLife/EveronLife#68] [Polls]
I would like to change the gathering system from the old Arma 3:
_"Stand still and press the same button over and over to get items"_

into a dynamic and random system that spawns gatherable items in an area:

![appleTree](https://user-images.githubusercontent.com/45827986/171422022-e3e72bed-2028-430f-8c55-2fabae37a430.gif)

![cropGrow](https://user-images.githubusercontent.com/45827986/171421744-e52e6d03-61c0-4151-968f-33be38c014be.gif)

**Comment by thebonbon** [comment]
Related to https://github.com/EveronLife/EveronLife/issues/7

**Comment by wrdg** [comment]
I see no issue with this, and figured it was implied since the limitation from A3 life was that everything was mission bound if you weren't playing on a server with mods loaded.

**Comment by Arkensor** [comment]
We can have both options. There are two problems with interactive item picking:
1. They require a lot of attention and interaction. When you are farming for hours this can become very exhausting. I was happy I could hold a button to gather and watch videos on a second monitor in the meantime.
2. From a technical standpoint this will be difficult to do for MP. Spawning individual gatherable nodes for each of the trees (in case of apple, ALL of them on the map) would create a massive amount of objects. We could try and spice things up clientside only. For trees in the visible range, we could show them and only validate on the server that someone must be near the source to pick an apple and that the interaction can not be faster than X gathered per second.

I think we should stick with the F-hold system for now until we figure out a solid way of doing more details without having the server manage all of it.

It is worth pointing out, that we will likely tweak the action to auto-repeat so you can just keep holding F until your inventory is full.

**Reply by thebonbon** [reply]
> 1. They require a lot of attention and interaction. When you are farming for hours this can become very exhausting. I was happy I could hold a button to gather and watch videos on a second monitor in the meantime.
> 
Thats the point, a more interactable gathering process. It is RP after all so why "afk" farm?
> 2. From a technical standpoint this will be difficult to do for MP. Spawning individual gatherable nodes for each of the trees (in case of apple, ALL of them on the map) would create a massive amount of objects. We could try and spice things up clientside only. For trees in the visible range, we could show them and only validate on the server that someone must be near the source to pick an apple and that the interaction can not be faster than X gathered per second.
> 
That is true it would spawn (at 10 apples per tree) a total of ~4000 apples and it starts lagging. But we have the new technology and pretty sure there is a way. (Like you said clientside etc..)
> I think we should stick with the F-hold system for now until we figure out a solid way of doing more details without having the server manage all of it.
> 
We can stick with it for now but I strongly disagree keeping it.
> It is worth pointing out, that we will likely tweak the action to auto-repeat so you can just keep holding F until your inventory is full.
>
That makes it in my opinion even worse.

**Reply by Arkensor** [reply]
Well, that is why we keep things adjustable. I am sure some people will prefer having either option for their server. As long as it does not cause too much work to maintain both, I think it's a good idea to just enable the users to choose what they like and not worry about trying to achieve general consensus. 

As long as the clientside sub-node gathering is not fully worked out we just go with the easier to implement alternative so we have some gathering system in place.
